src/Microsoft.AspNet.OData.Shared/Builder/ParameterConfiguration.cs

```diff
@@ -51,5 +51,44 @@ protected ParameterConfiguration(string name, IEdmTypeConfiguration parameterTyp
         /// Gets or sets a value indicating whether this parameter is nullable or not.
         /// </summary>
         public bool Nullable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this parameter is optional or not.
+        /// </summary>
+        public bool IsOptional { get; set; }
```

Comment:
Should we do a private set? It is possible to call HasDefaultValue and then set IsOptional to false. I don't think it causes any serious issues but I would be confused by the API.

Reply:
Yes. Thanks. I changed it to use "protected set".